gitapp/lib/users_list/home_page_view.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:flutter/material.dart';
+import 'package:gitapp/users_list/searchbar_widget.dart';
+import 'package:gitapp/constants/constants.dart';
+import 'package:gitapp/users_list/connection_presenter.dart';
+import 'package:gitapp/users_list/user_card_widget.dart';
+import 'package:gitapp/users_list/users_presenter.dart';
+import 'no_connection_widget.dart';
+
+class HomePage extends StatefulWidget {
+  HomePage({Key? key, required this.title}) : super(key: key);
+
+  final String title;
+
+  @override
+  HomePageState createState() => HomePageState();
+}
+
+class HomePageState extends State<HomePage> with TickerProviderStateMixin {
+  final UsersPresenter users = UsersPresenter();
+  late Searchbar searchBar = new Searchbar(getUsers);
+  bool connectionStatus = true;
+  bool searching = false;
+
+  Future getUsers(String text) async {
+    connectionStatus = await Connection.instance.checkConnection();
+    setState(() {});
+    if (text != "" || !connectionStatus) {
+      setState(() {
+        searching = true;
+      });
+      await users.fetchUsersFromApi(text);
+      setState(() {
+        searchBar.textController.clear();
+        searching = false;
+      });
+    } else
+      users.searchedList.clear();
+  }
+
+  Widget widgetPicker() {
+    if (!connectionStatus) {
+      return Center(
+        child: NetworkErrorView(
+            message: 'No Internet Connection.\nMind Trying again?'),
+      );
+    } else if (searching) {
+      return Center(
+        child: CircularProgressIndicator(),
+      );
+    } else
+      return ListView.builder(
```

Comment:
This ListView could be moved to another widget as a whole part.

Reply:
Solved